lib/index.js

```diff
@@ -1,6 +1,7 @@
 var Assert = require('assert');
 var Hoek = require('hoek');
 var Joi = require('joi');
+var Schema = require('mongoose').Schema
```

Comment:
If possible, please could we not `require` mongoose? I'm trying to keep Joigoose as flexible as possible, and not have any dependency on any particular version of mongoose.

You could declare `Schema` in `internals.root`, after `internals.mongoose` has been set.